metadata-ingestion/src/datahub/ingestion/source/sql/postgres.py

```diff
@@ -144,13 +156,12 @@ def create(cls, config_dict, ctx):
         return cls(config, ctx)
 
     def get_inspectors(self) -> Iterable[Inspector]:
-        # This method can be overridden in the case that you want to dynamically
-        # run on multiple databases.
-        url = self.config.get_sql_alchemy_url()
+        # init_database is overwritten in url if database or sqlalchemy_uri are provided
+        url = self.config.get_sql_alchemy_url(database=self.config.init_database)
```

Comment:
imo this is super confusing - if you pass `database` explicitly, it still may not be respected in some circumstances? I think we should change BasicSQLAlchemyConfig's method to prioritize the kwarg first and only use the default value after + make the change below

```suggestion
        url = self.config.get_sql_alchemy_url(database=self.config.database or self.config.init_database)
```

Reply:
That works, was just trying to limit changes / keep existing functionality